### PR TITLE
Refactor AbilityStrategy passives & synergies

### PR DESCRIPTION
**Pull request by Exeatop**

### CODE REVIEW

File: app/core/abilities/ability-strategy.ts [hunk 1]
```diff
@@ -1,15 +1,16 @@
...
-import { Item } from "../../types/enum/Item"
 import { Passive } from "../../types/enum/Passive"
 import { Synergy } from "../../types/enum/Synergy"
 import { distanceC } from "../../utils/distance"
 import Board from "../board"
 import { PokemonEntity } from "../pokemon-entity"
 import PokemonState from "../pokemon-state"
-import { AbilityStrategies } from "./abilities"
 import { min } from "../../utils/number"
+import { triggerItems } from "../items"
```
Comment (Phoenix-Starlight) [inline app/core/abilities/ability-strategy.ts:10]: `triggerItemsEffects` intended instead?

File: app/core/items.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        board: Board,
+        target: PokemonEntity,
+        crit: boolean
+}){
+    params.pokemon.items.forEach((i) =>{
+        const handler = itemMap.get(i)
+        if(handler){
+            handler(effect_class, params)
+        } else{
+            console.error(`${i} missing from itemMap`)
```
Comment (Phoenix-Starlight) [inline app/core/items.ts]: There is a logger for this.

File: app/core/abilities/ability-strategy.ts (hunk 1 above)
Comment (Exeatop) [reply app/core/abilities/ability-strategy.ts:10]: My items file wasn't updated properly. triggerItems is the current name of the routing function

File: app/core/items.ts (hunk 2 above)
Comment (Exeatop) [reply app/core/items.ts]: Yup was just me checking it. File didn't update properly

File: app/core/passives.ts [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    const pokemon = params.pokemon
+    switch(effect_class){
+        case EffectClass.ALLY_SOUND_SYNERGY_ON_ABILITY:
+            const ally = params.target
+            if(distanceC(
+                pokemon.positionX,
+                pokemon.positionY,
+                ally.positionX,
+                ally.positionY
+            )<=1){
```
Comment (Phoenix-Starlight) [inline app/core/passives.ts:109]:
```suggestion
            ) <= 1 ) {
```

Formatting

File: app/core/synergies.ts [hunk 4]
```diff
@@ -0,0 +1,94 @@
+import { EffectClass } from "../types/enum/EffectClass";
+import {PokemonEntity} from "./pokemon-entity"
```
Comment (Phoenix-Starlight) [inline app/core/synergies.ts:2]: Formatting here

**Review by Phoenix-Starlight — CHANGES_REQUESTED** [review]
Pointed out some formatting issues, but go through the files. Additionally Biome has a formatter.

**Review by Phoenix-Starlight — COMMENTED** [review]
The switches on enum seem redundant.

**Comment by Exeatop** [comment]
> The switches on enum seem redundant.

The idea is to concentrate each feature, but features' effects must be triggered in different circumstances so the idea of the EffectClass type was to be able to distinguish between different contexts.

**Pull request closed without merging**